sdk/rust/oak/src/io/mod.rs

```diff
@@ -87,6 +89,47 @@ pub fn send_init<Init: Encodable + Decodable, Command: Encodable + Decodable>(
     Ok(command_sender)
 }
 
+/// Sends [`crate::grpc::Invocation`] (containing [`oak_io::Receiver`] and [`oak_io::Sender`])
+/// through invocation sender if [`oak_io::Receiver`] label flows to invocation sender's label.
+/// If failed - sends error back through [`oak_io::Sender`].
+///
+/// Useful for sending invocations from router nodes and checking label correctness without actually
+/// reading the contents of invocation.
+pub fn forward_invocation(
+    invocation: crate::grpc::Invocation,
+    invocation_sender: &Sender<crate::grpc::Invocation>,
+) -> anyhow::Result<()> {
+    match (&invocation.receiver, &invocation.sender) {
+        (Some(request_receiver), Some(response_sender)) => {
+            let request_label = request_receiver
+                .label()
+                .context("Couldn't read request label")?;
+            let sender_label = crate::channel_label_read(invocation_sender.handle.handle)
+                .context("Couldn't read invocation sender label")?;
+            // Check if request label is valid in the context of invocation sender.
+            if !request_label.flows_to(&sender_label) {
+                let grpc_response_writer =
+                    crate::grpc::ChannelResponseWriter::new(response_sender.clone());
+                grpc_response_writer
+                    .close(Err(crate::grpc::build_status(
+                        rpc::Code::InvalidArgument,
+                        "Invalid request label",
+                    )))
+                    .context("Couldn't send error response back")?;
+                anyhow::bail!("Invalid request label: {:?}", request_label);
+            }
+
+            // Forward invocation through invocation sender.
+            invocation_sender
+                .send(&invocation)
+                .context("Couldn't forward invocation")
+        }
+        _ => {
+            anyhow::bail!("Received malformed invocation");
```

Comment:
Maybe just return `Err(anyhow::anyhow!("..."))`, the early return is not necessary here.

Reply:
Done